Internals/Common/Input.cs

```diff
@@ -20,6 +20,9 @@ public static class Input
         public static GamePadState OldGamePadSnapshot { get; internal set; }
 
         public static void HandleInput(PlayerIndex pIndex = PlayerIndex.One) {
+            OldKeySnapshot = CurrentKeySnapshot;
+            OldMouseSnapshot = CurrentMouseSnapshot;
+            OldGamePadSnapshot = CurrentGamePadSnapshot;
```

Comment:
have you tested that moving these here works?

Reply:
It works, prior to this my checks in `WPTR.DoRender` for `UIElement` mouse events would not work when using the newly added `Input.OldMouseX` but does now.